pkg/trace/config/apply.go

```diff
@@ -142,24 +142,28 @@ func (c *AgentConfig) applyDatadogConfig() error {
 			log.Infof("'site' and 'apm_dd_url' are both set, using endpoint: %q", host)
 		}
 	}
-	for url, keys := range config.Datadog.GetStringMapStringSlice("apm_config.additional_endpoints") {
-		if len(keys) == 0 {
-			log.Errorf("'additional_endpoints' entries must have at least one API key present")
-			continue
-		}
-		for _, key := range keys {
-			c.Endpoints = append(c.Endpoints, &Endpoint{Host: url, APIKey: key})
+	if config.Datadog.IsSet("apm_config.additional_endpoints") {
+		for url, keys := range config.Datadog.GetStringMapStringSlice("apm_config.additional_endpoints") {
```

Comment:
You can simply set a default value of `[]string{}` in `config.go` for this option to remove the option, see https://github.com/DataDog/datadog-agent/commit/3e46529aacb3048a9a95fe3c3e78709d62c4004d#diff-091ace02b49a46f55f2586cfec89a160

For `proxy.no_proxy` I'd sync with croissant, IIRC there is a utility method to access this, instead of directly accessing the fields

Reply:
Happy to do that, but I think we should be good like this for now. I'd rather not introduce too many changes in this PR and risk bringing up other problems now during the release week.

Let me know if you think otherwise and we can reconsider.

Reply:
> You can simply set a default value of []string{} in config.go for this option to remove the option, see 3e46529#diff-091ace02b49a46f55f2586cfec89a160

I'm not sure if having a default here actually provides any benefit. I think this way is more correct - simply checking if the value is set.

> For proxy.no_proxy I'd sync with croissant, IIRC there is a utility method to access this, instead of directly accessing the fields

I didn't manage to find any utility on my own, but I'd rather not add one at this point and risk introduce some incompatibility or other issue.

Let's just solve the issue at hand directly for now and think about potential improvements later, what do you think?